src/main/java/gr/cytech/chatreminderbot/rest/BotResource.java

```diff
@@ -42,50 +49,45 @@ public String handleReq(Request req) {
             logger.warn("Error from message:{}", req.getMessage().getText(), e);
             message = "Not even a clue what you just said";
         }
+
+        //case message is already build in json
+        if (message.startsWith("{\"actionResponse\":{\"type\":")) {
+            return message;
+        }
         return responseBuild(spaceId, message);
     }
 
-    @GET
-    @Path("/button")
-    public String onButtonClick(@Context HttpServletRequest request) {
-        Sender sender = new Sender();
-        ThreadM threadM = new ThreadM();
-
-        sender.setName(request.getParameter("name"));
-        threadM.setName("space/" + request.getParameter("space") + "/thread/" + request.getParameter("thread"));
-
-        //create reminder and get the when
-        //creating the full text for reminder
-        String text = "remind me "
-                + request.getParameter("text")
-                + " in 10 minutes";
-
-        //create the Request using the updated message
-        Message message = new Message();
+    private void manipulateRequestBasedOnParameters(Request req) {
+        String text = "";
+        int reminderId = 0;
+        for (int i = 0; i < req.getAction().getParameters().size(); i++) {
+            String tempValue = req.getAction().getParameters().get(i).get("value");
+            String tempKey = req.getAction().getParameters().get(i).get("key");
+            if ("text".equals(tempKey)) {
+                text = tempValue;
+            } else if ("reminderId".equals(tempKey)) {
+                reminderId = Integer.valueOf(tempValue);
+            }
+        }
 
-        message.setSender(sender);
-        message.setThread(threadM);
-        message.setText(text);
+        req.getMessage().getSender().setName(req.getUser().getName());
 
-        Request req = new Request();
-        req.setMessage(message);
-        //open tab to get the requirements then immediately close it and handle the request
-        handleReq(req);
-        return "<html>"
-                + "<head></head>"
-                + "<body>"
-                + "<script>"
-                + "window.close();"
-                + "</script>"
-                + "</body>"
-                + "</html>";
+        if (REMIND_AGAIN_IN_10_MINUTES.equals(req.getAction().getActionMethodName())) {
+            req.getMessage().setText("remind me " + text + " in 10 minutes");
```

Comment:
can all these hard-coded strings be replaced by something more type-safe and not-so-error-prone?